File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    moved = []

    for i in moves:
        for j in range(len(board)):
            if board[j][i-1] != 0:
                moved.append(board[j][i-1])
                board[j][i-1] = 0
                break
        if len(moved) >= 2 and moved[len(moved)-1] == moved[len(moved)-2]:
            moved.pop(-1)
            moved.pop(-1)
            answer += 2

    return answer

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        List<Integer> moved = new ArrayList<Integer>();
        
        for(int m : moves) {
            for(int i=0; i<board.length; i++) {
                if(board[i][m-1] != 0) {
                    moved.add(board[i][m-1]);
                    board[i][m-1] = 0;
                    break;
                }
            }
            if(moved.size() >= 2 && moved.get(moved.size()-1) == moved.get(moved.size()-2) ) {
                moved.remove(moved.size()-1);
                moved.remove(moved.size()-1);
                answer += 2;
                
            }
        }
        
        return answer;
    }
}